### PR TITLE
Update auth helpers sveltekit guide to match the latest release

### PR DESCRIPTION
**Pull request by silentworks**
## What kind of change does this PR introduce?

Guide update for SvelteKit Auth Helpers

## What is the current behavior?

Please link any relevant issues here.

## What is the new behavior?

Feel free to include screenshots if it includes visual changes.

## Additional context

Add any other context or screenshots.

### CODE REVIEW

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx [hunk 1]
```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
...
 ### Set up the Supabase client
 
-Start off by creating a `db.ts` file inside of the `src/lib` directory and instantiate the `supabaseClient`.
-
-```ts title=src/lib/db.ts
-import { createClient } from '@supabase/auth-helpers-sveltekit'
-import { env } from '$env/dynamic/public'
-// or use the static env
-// import { PUBLIC_SUPABASE_URL, PUBLIC_SUPABASE_ANON_KEY } from '$env/static/public';
+Create the server client in a handle hook:
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:59]: Not sure what a server client is?

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx [hunk 2]
```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
...
 
-  onMount(() => {
+export const handle: Handle = async ({ event, resolve }) => {
+  event.locals.supabase = createSupabaseServerClient({
+    supabaseUrl: PUBLIC_SUPABASE_URL,
+    supabaseKey: PUBLIC_SUPABASE_ANON_KEY,
+    event
+  });
+
+  // a little helper to save some lines
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:77]: If this needs a comment, I'd maybe add what it does. What lines are we saving?

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx [hunk 3]
```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
...
+};
 ```
 
-In addition you can create a layout load function if you are using `invalidate('supabase:auth')`:
+#### Shared Load functions and pages
 
-```ts title=src/routes/+layout.ts
-import type { LayoutLoad } from './$types'
-import { getSupabase } from '@supabase/auth-helpers-sveltekit'
+To be able to use supabase in shared load functions and inside pages you need to create a supabase client in the root layout load:
```
[the remark this answers is not in the document]
Comment (saltcod) [reply apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:117]: Also, what's a shared load function? Sveltekit specific that everyone would know?

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx [hunk 4]
```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
...
+  const {
+    data: { session }
+  } = await supabase.auth.getSession();
+
+  return { supabase, session };
+};
 ```
 
-This results in fewer server calls as the client manages the session on its own.
+The client can be accessed inside pages by `$page.data.supabase` or `data.supabase` when using `export let data: PageData`.
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:151]: "Access the client inside pages by....

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx [hunk 5]
```diff
@@ -42,211 +49,239 @@ yarn add @supabase/auth-helpers-sveltekit
...
 
-### Typings
+The usage of `depends` and `invalidate` tells sveltekit that this load function should be executed whenever the session updates to keep the page store in sync.
 
-In order to get the most out of TypeScript and it´s intellisense, you should import our types into the `app.d.ts` type definition file that comes with your SvelteKit project.
+`createSupabaseLoadClient` caches the client when running in a browser environment and therefore does not create a new client for every time the load function runs.
 
-```ts title=src/app.d.ts
-/// <reference types="@sveltejs/kit" />
+### Typings with v2 from Supabase CLI
```
Comment (saltcod) [inline apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:157]: "Generate types from your database"  ?

Note that you can also download them right from the table now (API Preview) button

**Review by saltcod — APPROVED** [review]

File: apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx (hunk 3 above)
[the remark this answers is not in the document]
Comment (silentworks) [reply apps/docs/pages/guides/auth/auth-helpers/sveltekit.mdx:117]: > Also, what's a shared load function? Sveltekit specific that everyone would know?

Yes this is SvelteKit specific and anyone using SvelteKit would know this or they should revert to the SvelteKit docs for this.

**Pull request merged**